File: Backend-main/src/contact/contact.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { ContactFormEntity } from 'src/entities/ContactFormEntity';
import { Repository } from 'typeorm';
import { ContactFormRequest, ContactFormStatus } from './interface/contact.interface';

@Injectable()
export class ContactService {
  constructor(
    @InjectRepository(ContactFormEntity)
    private readonly contactFormRepository: Repository<ContactFormEntity>,
  ) {}

  async createContactFormEntry(request: ContactFormRequest, userId?: string): Promise<void> {
    await this.contactFormRepository.insert({ ...request, userId });
  }

  async getOpenContactForms(): Promise<ContactFormEntity[]> {
    return this.contactFormRepository.find({ where: { status: ContactFormStatus.OPEN } });
  }

  async getContactFormById(id: string): Promise<ContactFormEntity | null> {
    return this.contactFormRepository.findOne({ where: { id } });
  }

  async closeContactForm(adminId: string, id: string): Promise<void> {
    await this.contactFormRepository.update(
      { id },
      { status: ContactFormStatus.CLOSED, closedBy: adminId },
    );
  }
}

File: spring-backend/src/main/java/com/simmortal/contact/ContactService.java
package com.simmortal.contact;

import org.springframework.stereotype.Service;

@Service
public class ContactService {
  private final ContactRepository contactRepository;

  public ContactService(ContactRepository contactRepository) {
    this.contactRepository = contactRepository;
  }

  public void createContactFormEntry(ContactFormRequest request, String userId) {
    contactRepository.createContactForm(request, userId);
  }

  public Object getOpenContactForms() {
    return contactRepository.getOpenContactForms();
  }

  public Object getContactFormById(String id) {
    return contactRepository.getContactFormById(id);
  }

  public Object closeContactForm(String adminUserId, String id) {
    return contactRepository.closeContactForm(adminUserId, id);
  }
}
